dual_screen/android/src/main/kotlin/com/builttoroam/dual_screen/DualScreenPlugin.kt

```diff
@@ -0,0 +1,132 @@
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+/** DualScreenPlugin */
+public class DualScreenPlugin :
+    FlutterPlugin,
+    MethodCallHandler,
+    EventChannel.StreamHandler,
+    ActivityAware {
+    private val METHOD_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/methods"
+    private val EVENT_CHANNEL_NAME = "plugins.builttoroam.com/dual_screen/events"
+    private lateinit var methodChannel: MethodChannel
+    private lateinit var eventChannel: EventChannel
+    private var context: Context? = null
+    private var activity: Activity? = null
+
+    fun registerPlugin(context: Context?, messenger: BinaryMessenger?) {
+        activity = this.activity
+        this.context = context
+        MethodChannel(messenger, METHOD_CHANNEL_NAME).setMethodCallHandler(this)
+        EventChannel(messenger, EVENT_CHANNEL_NAME).setStreamHandler(this)
+    }
+
+    override fun onAttachedToEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        this.registerPlugin(binding.getApplicationContext(), binding.getBinaryMessenger())
+    }
+
+    companion object {
+        @JvmStatic
+        fun registerWith(registrar: Registrar) {
+            DualScreenPlugin().registerPlugin(registrar.context(), registrar.messenger())
+        }
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            "isDualScreenDevice" -> {
+                result.success(isDualScreenDevice())
+            }
+            "isAppSpanned" -> {
+                if (isDualScreenDevice()) {
+                    result.success(isAppSpanned())
+                } else {
+                    result.success(false)
+                }
+            }
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    override fun onListen(arguments: Any?, events: EventSink?) {
+        activity?.getWindow()?.getDecorView()?.getRootView()?.addOnLayoutChangeListener(object : OnLayoutChangeListener {
+            override fun onLayoutChange(
+                v: View?,
+                left: Int,
+                top: Int,
+                right: Int,
+                bottom: Int,
+                oldLeft: Int,
+                oldTop: Int,
+                oldRight: Int,
+                oldBottom: Int
+            ) {
+                events!!.success(isAppSpanned())
+            }
+        })
+    }
+
+    override fun onCancel(arguments: Any?) {
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        this.activity = binding.activity
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        this.activity = null
```

Comment:
I'm not sure of the purpose of these two calls that are nullifying the activity but should we do similar for context? That is If this is part of the disposing process.

Reply:
My thoughts for only including activity here were that this method is only related to activity changes and not context changes